lib/internal/modules/cjs/loader.js

```diff
@@ -1909,6 +1918,12 @@ Module._extensions['.js'] = function(module, filename) {
  * @param {string} filename The file path of the module
  */
 Module._extensions['.json'] = function(module, filename) {
+
+  const isAllowedToExecute = ci.isAllowedToExecuteFile(filename);
```

Comment:
As I said earlier. It would be better placed inside Module._load so you don't need to replicate it to every extension.